Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -3297,15 +3297,97 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
         }
     }
 
-    void vectorLoad8Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad16Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad32Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad64Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-
-    void vectorLoad8Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad16Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad32Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad64Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
+    void vectorLoad8Splat(Address address, FPRegisterID dest, FPRegisterID scratch)
+    {
+        if (supportsAVXForSIMD()) {
+            m_assembler.vpinsrb_mrr(0, address.offset, address.base, dest, dest);
+            m_assembler.vpxor_rrr(scratch, scratch, scratch);
+            m_assembler.vpshufb_rrr(scratch, dest, dest);
+        } else
+            RELEASE_ASSERT_NOT_REACHED();
+    }
+
+    void vectorLoad16Splat(Address address, FPRegisterID dest)
+    {
+        if (supportsAVXForSIMD()) {
+            m_assembler.vpinsrw_mrr(0, address.offset, address.base, dest, dest);
+            m_assembler.pshuflw_rr(0, dest, dest);
+            m_assembler.punpcklqdq_rr(dest, dest);
+        } else
```

Comment:
Define and use AVX functions.

Reply:
For other AVX functions, I will implement them in another patch.